src/twisted/web/test/test_xmlrpc.py

```diff
@@ -661,17 +663,18 @@ def setUp(self):
 
 
     def test_authInfoInURL(self):
-        url = "http://%s:%s@127.0.0.1:%d/" % (
-            nativeString(self.user), nativeString(self.password), self.port)
-        p = xmlrpc.Proxy(networkString(url))
+        url = u"http://{}:{}@127.0.0.1:{}/".format(
+            self.user.decode("utf-8"), self.password.decode("utf-8"),
+            self.port)
+        p = xmlrpc.Proxy(url.encode("utf-8"))
```

Comment:
Why is this URL encoded as UTF-8, when the others are encoded as ASCII?

The best thing to do would be to use `hyperlink.URL.from_text`, then `to_url()`, then encode _that_ as ASCII.  That will encode each component of the URL according to its type.  @mahmoud Can you confirm this?